### PR TITLE
Data stores

### PR DESCRIPTION
**Pull request by jpopelka**

### CODE REVIEW

File: data-stores/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Why
+
+At the moment (Nov 19) we use Redis as a Celery backend (i.e. for storing results of Celery tasks) as well as a data store where we track copr builds, github app installations and whitelist of users.
+Redis is in-memory DB, i.e. the more data we store the more memory we use.
+Because memory is (unlike disk) quite expensive we want to move to a DB which stores the data on disk.
+The task here is to decide between SQL or NoSQL and then between our own deployment or a database in a cloud.
```
Comment (TomasTomecek) [inline data-stores/README.md]: +1 very nice write-up

File: data-stores/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+# SQL
+
+SQL stores data in a structured way of interconnected tables.
+The question here is whether we actually need structured tables to do some crazy queries.
```
Comment (TomasTomecek) [inline data-stores/README.md]: Right now: not really
In future (as the service grows more complex): probably

File: data-stores/README.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+Big plus here is that Celery supports [SQLAlchemy](https://www.sqlalchemy.org) as a [build-in backend](https://docs.celeryproject.org/en/latest/getting-started/first-steps-with-celery.html#keeping-results) and the same applies to [Flask](https://github.com/pallets/flask-sqlalchemy).
```
Comment (TomasTomecek) [inline data-stores/README.md]:
```suggestion
Big plus here is that Celery supports [SQLAlchemy](https://www.sqlalchemy.org) as a [built-in backend](https://docs.celeryproject.org/en/latest/getting-started/first-steps-with-celery.html#keeping-results) and the same applies to [Flask](https://github.com/pallets/flask-sqlalchemy).
```

File: data-stores/README.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+### Cons
+Because SQLite is a serverless database, it doesn’t provide direct network access to its data. The application (as I understand it) just stores data in a file by using a library. If more containers (in our case service and more workers) need to access the db, they probably need to have the file on a shared (RWX) volume.
+
+## Postgresql
+
+### Pros
+[JSON data type](http://www.postgresqltutorial.com/postgresql-json) is a big plus, because we already have our data serialized as jsons so we can just store them that way and then do all kinds of queries over them.
+
+### Cons
+Overkill
```
Comment (TomasTomecek) [inline data-stores/README.md]: not really, as we grow bigger, psql can easily scale with our needs

File: data-stores/README.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+# Why
+
+At the moment (Nov 19) we use Redis as a Celery backend (i.e. for storing results of Celery tasks) as well as a data store where we track copr builds, github app installations and whitelist of users.
```
Comment (TomasTomecek) [inline data-stores/README.md]:
```suggestion
At the moment (Oct 19) we use Redis as a Celery backend (i.e. for storing results of Celery tasks) as well as a data store where we track copr builds, github app installations and whitelist of users.
```
Are you time-travelling?

File: data-stores/README.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+## Pros:
+Stores data in flexible, JSON-like documents, meaning fields can vary from document to document and data structure can be changed over time. It also [supports queries](https://www.tutorialspoint.com/mongodb/mongodb_query_document.htm) and indexing.
+## Cons
+Using MongoDB as a Celery backend is not that straightforward, but there are options: [[1]](http://docs.celeryproject.org/en/latest/_modules/celery/backends/mongodb.html), [[2]](https://stackoverflow.com/questions/15740755/working-example-of-celery-with-mongo-db), [[3]](https://stackoverflow.com/questions/53017827/example-celery-v4-2-with-mongodb-results-backend)
+
+# Cloud
+
+Deploy or cloud-native? That's the question.
+
+## Deploy in OpenShift
```
Comment (TomasTomecek) [inline data-stores/README.md]: Yes, please.

File: data-stores/README.md (hunk 5 above)
Comment (jpopelka) [reply data-stores/README.md]: No, I'm not, are you? My desktop says Mon Nov 4

File: data-stores/README.md (hunk 6 above)
Comment (jpopelka) [reply data-stores/README.md]: Yes please what?

File: data-stores/README.md (hunk 4 above)
Comment (jpopelka) [reply data-stores/README.md]: MongoDB can scale even better:
- https://www.digitalocean.com/community/tutorials/a-comparison-of-nosql-database-management-systems-and-models#relational-databases-and-their-limitations
- https://www.digitalocean.com/community/tutorials/a-comparison-of-nosql-database-management-systems-and-models#document-oriented-databases

File: data-stores/README.md (hunk 5 above)
Comment (TomasTomecek) [reply data-stores/README.md]: It will be valid tomorrow!

File: data-stores/README.md (hunk 6 above)
Comment (TomasTomecek) [reply data-stores/README.md]: deploy > cloud native

otherwise you'll be the one to convince our management that we need to pay for another thing

**Pull request merged**